Integrations/Digital_Defense_FrontlineVM/Digital_Defense_FrontlineVM.py

```diff
@@ -0,0 +1,795 @@
+    query_date = datetime.strftime(query_date, "%Y-%m-%dT%H:%M:%SZ")
+    return query_date
+
+
+def get_fetch_frequency_td():
+    ''' Returns the INCIDENT_FREQUENCY as a datetime.timedelta object    '''
+    fetch_frequency = INCIDENT_FREQUENCY.split()
+    demisto.debug("FrontlineVM get_fetch_incident_td -- using frequency: " + str(fetch_frequency))
+    if "min" in str(fetch_frequency[1]):
+        return timedelta(minutes=int(fetch_frequency[0]))
+    return timedelta(hours=int(fetch_frequency[0]))
+
+
+def create_vuln_event_object(vuln):
+    ''' Creates a vulnerability event object given raw vulnerability data.  '''
+    vuln_event = {}
+    vuln_event['vuln-id'] = vuln.get('id')
+    vuln_event['hostname'] = vuln.get('hostname')
+    vuln_event['ip-address'] = vuln.get('ip_address')
+    vuln_event['port'] = vuln.get("port")
+    vuln_event['scan-id'] = vuln.get('scan_id')
+    vuln_event['vuln-title'] = vuln.get('title')
+    vuln_event['date-created'] = vuln.get('active_view_date_created')
+    vuln_event['ddi-severity'] = vuln['severities']['ddi']
+    vuln_event['vuln-info'] = vuln.get('data')
+    return vuln_event
+
+
+def vulns_to_incident(vulns, start_time_dt):
+    '''
+        Iterate through vulnerabilities and create incident if
+        vulnerability has been created since last start_time.
+    '''
+    incidents = []
+    for vuln in vulns:
+        # get vulnerability active view (av) date created values:
+        av_date_created_str = vuln.get('active_view_date_created')
+        av_date_created_dt = datetime.strptime(av_date_created_str, "%Y-%m-%dT%H:%M:%S.%fZ")
+
+        # Skip if vuln created before current start_time:
+        if av_date_created_dt < start_time_dt:
+            continue
+        vuln_event = create_vuln_event_object(vuln)
+        incident = {
+            'name': vuln.get('title'),
+            'occurred': vuln.get('active_view_date_created'),
+            'details': vuln.get('data'),
+            'rawJSON': json.dumps(vuln_event)
+        }
+        incidents.append(incident)
+    return incidents
+
+
+def fetch_incidents():
+    ''' Method to fetch Demisto incidents by pulling any new vulnerabilities found.   '''
+    try:
+        last_run = demisto.getLastRun()
+        # Check if last_run exists and has a start_time, else create new start_time timestamp:
+        if last_run and last_run.get('start_time', False):
+            start_time = last_run.get('start_time')
+            # Check if user selected fetch frequency is within last_run[start_time]:
+            fetch_frequency_td = get_fetch_frequency_td()
+            if (datetime.utcnow() - datetime.utcfromtimestamp(int(start_time))) < fetch_frequency_td:
+                debug_msg = "frequency (" + str(fetch_frequency_td) + ") is within last start_time, sending empty incidents."
+                demisto.debug("FrontlineVM fetch_incidents -- " + debug_msg)
+                demisto.setLastRun({'start_time': last_run.get('start_time')})
```

Comment:
let's delete this statement, it changes nothing in the last run data

Reply:
Commented out for now. 
Will test with commented out. 
I remember adding that for a reason, can't remember why. I think the last run time was not staying in memory unless I re-set it before sending an "empty" incident.